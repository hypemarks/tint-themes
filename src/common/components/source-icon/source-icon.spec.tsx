import { render } from '@testing-library/react';
import React from 'react';
import SourceIcon from './source-icon';

describe('Test <Icon /> Component', () => {
  it('should match snapshot', () => {
    const wrapper = render(<SourceIcon type={'tiktok'} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should create source icon with correct classes', () => {
    const wrapper = render(<SourceIcon type={'tiktok'} isNavItem={true} />);
    const icon = wrapper.queryByTestId('source-icon');
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass('source-icon--nav-item');
  });
});
