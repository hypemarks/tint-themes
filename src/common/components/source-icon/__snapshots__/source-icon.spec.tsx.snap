// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Icon /> Component should match snapshot 1`] = `
<DocumentFragment>
  <img
    alt="tiktok icon"
    class="source-icon source-icon--image"
    data-testid="source-icon"
    src="https://cdn.hypemarks.com/assets/profile/tiktok.svg"
  />
</DocumentFragment>
`;
